aws_lambda_powertools/utilities/parameters/secrets.py

```diff
@@ -224,3 +359,88 @@ def get_secret(
         force_fetch=force_fetch,
         **sdk_options,
     )
+
+
+def set_secret(
+    name: str,
+    value: Union[str, bytes],
+    *,  # force keyword arguments
+    client_request_token: Optional[str] = None,
+    **sdk_options,
+) -> str:
+    """
+    Modify the details of a secret or create a new secret if it doesn't already exist.
+    It includes metadata and the secret value.
+
+    We aim to minimize API calls by assuming that the secret already exists and needs updating.
+    If it doesn't exist, we attempt to create a new one. Refer to the following workflow for a better understanding:
+
+
+                      ┌────────────────────────┐      ┌─────────────────┐
+            ┌───────▶│Resource NotFound error?│────▶│Create Secret API│─────┐
+            │         └────────────────────────┘      └─────────────────┘     │
+            │                                                                 │
+            │                                                                 │
+            │                                                                 ▼
+    ┌─────────────────┐                                              ┌─────────────────────┐
+    │Update Secret API│────────────────────────────────────────────▶│ Return or Exception │
+    └─────────────────┘                                              └─────────────────────┘
+
+    Parameters
+    ----------
+    name: str
+        The ARN or name of the secret to add a new version to or create a new one.
+    value: str, dict or bytes
+        Specifies text data that you want to encrypt and store in this new version of the secret.
+    client_request_token: str, optional
+        This value helps ensure idempotency. Recommended that you generate
+        a UUID-type value to ensure uniqueness within the specified secret.
+        This value becomes the VersionId of the new version. This field is
+        auto-populated if not provided.
+    sdk_options: dict, optional
+        Dictionary of options that will be passed to the Secrets Manager update_secret API call
+
+    Raises
+    ------
+    SetSecretError
+        When attempting to update or create a secret fails.
+
+    Returns:
+    -------
+    SetSecretResponse:
+        The dict returned by boto3.
```

Comment:
The function signature is returning a `str`. Which one is correct?

Reply:
Leftover, fixed!